lib/facter/self_service.rb

```diff
@@ -114,6 +114,81 @@
     { S0014: res.nil? }
   end
 
+  chunk(:S0016) do
+    # Puppetserver
+    next unless PuppetSelfService.primary? || PuppetSelfService.compiler? || PuppetSelfService.legacy_compiler?
+    log_path = Puppet.settings['logdir'].to_s + '/../puppetserver/puppetserver.log'
```

Comment:
`File.dirname` can be used instead of the `../`

```suggestion
    log_path = File.dirname(Puppet.settings['logdir'].to_s) + '/puppetserver/puppetserver.log'
```